File: scripts_go/functions_CPU.go
/*
 * The following functions test the cpu (quota cpu)
 */

package main

import (
	"math/rand"

	"github.com/anexia-it/go-e5e"
)

type entrypoints struct{}

func multiply(number int) {
	var result = 2
	for i := 0; i < number; i++ {
		result *= rand.Intn(999)
	}
}

// Test CPU by multiplying random numbers 100 times
func (f *entrypoints) Func_multiply_100(event e5e.Event, context e5e.Context) (*e5e.Return, error) {
	multiply(100)

	return &e5e.Return{
		Data: "Hello world!",
	}, nil
}

// Test CPU by multiplying random numbers 100.000 times
func (f *entrypoints) Func_multiply_100000(event e5e.Event, context e5e.Context) (*e5e.Return, error) {
	multiply(100000)

	return &e5e.Return{
		Data: "Hello world!",
	}, nil
}

// Test CPU by multiplying random numbers 1.000.000 times
func (f *entrypoints) Func_multiply_1000000(event e5e.Event, context e5e.Context) (*e5e.Return, error) {
	multiply(1000000)

	return &e5e.Return{
		Data: "Hello world!",
	}, nil
}

func main() {
	if err := e5e.Start(&entrypoints{}); err != nil {
		panic(err)
	}
}

File: scripts_ruby/functions_CPU.rb
#
# The following functions test the cpu (quota cpu)
#

def multply(number)
    result = 2

    for i in 1..number
        result *= rand(2..999)
    end
end

# Test CPU by multiplying 100 times
def func_multiply_100(event, context)
    multiply(100)

    return {
        :status => 200,
        :data => 'OK',
    }
end

# Test CPU by multiplying 100.000 times 
def func_multiply_100000(event, context)
    multiply(100000)

    return {
        :status => 200,
        :data => 'OK',
    }
end


# Test CPU by multiplying 1.000.000 times 
def func_multiply_1000000(event, context)
    multiply(1000000)

    return {
        :status => 200,
        :data => 'OK',
    }
end